waiter/src/waiter/metrics.clj

```diff
@@ -388,12 +398,30 @@
           (recur))
         (log/info "[transient-metrics-gc] stopping populating service-id->state-chan")))))
 
+(defn update-last-request-time
+  "Updates the last-request-time epoch time in the service-id->local-metrics to the maximum
+   of the current last request time and the provided last-request-datetime in epoch millis
+   for the specified service."
+  [service-id->local-metrics service-id ^DateTime candidate-last-request-time]
+  (let [current-last-request-time (get-in service-id->local-metrics [service-id "last-request-time"])
+        candidate-last-request-time-ms (.getMillis candidate-last-request-time)]
```

Comment:
I don't think we should adopt long millis as our representation of time.  We use `clj-time` for other timestamps, like `last-update-time` in the scheduler.  Using millis is going to make inspecting state inspection harder for humans.

Reply:
We should fix `last-update-time` in tokens to be consistent as well before it's too late.